### ISSUE TITLE
Could not process error because of unrelated file

### ISSUE THREAD

**Issue by BernhardWebstudio** [rectorphp/rector#1319]
Thank you very much for this utility! 
I am having a little issue with the `symfony-constructor-injection` level. 

**Rector Version**: v4.10 as well as dev-master@a65c7f2

**Command** I execute: 
`./bin/rector process ./src/ --autoload-file ./vendor/autoload.php --level symfony-constructor-injection --dry-run`
(I also tried without dry-run & without autoload-file, but I get the same error)

**Error** I get (multiple times, for different controllers but the same service & validator):
```
[ERROR] Could not process                                                                                              
         "/myDirectory/src/Acme/MyBundle/Controller/GeneralController.php" file, due to:                                                                                           
         "Cannot autowire service "Acme\MyOtherBundle\Validator\Constraints\AuthorizationValidator": argument "$security" of
         method "__construct()" references class "Symfony\Component\Security\Core\Security" but no such service         
         exists.".
...
```

While, when I check for the service with `bin/console debug:autowiring`, the service shows up fine (the app works fine the way it is, phpstan does not see any problems either):

 ```
Helper class for commonly-needed security tasks.
 Symfony\Component\Security\Core\Security (security.helper)
```

May I ask for some guidance on how to resolve this issue, what I am missing or how to debug rector's symfony autowiring?

**Comment by TomasVotruba** [comment]
Hi, this is very hard to debug without full application.
I could give you random tips, but it would only lead us to comment ping-pong.

Could you provide minimal application on Github with as few packages and files as possible, that still gives you this error?

**Comment by BernhardWebstudio** [comment]
Hi, thanks for your response. The easy fix for me was just to comment the classes requiring the security service, then it worked alright. 
Still curious (after all, when the problem is in context of security, it may be kind of relevant), I tried what you suggested to create a minimum viable application. Slowly deleting one file after another, I found that the error is related to the Symfony security configuration. My project had separate security configurations for production & debug. Apparently, rector may not be able to decide which environment it's run in. I guess this is a no-fix? Or a bug in an under-rector-lying library?

Anyway, if you want to reproduce, one may setup a Symfony app, create some service requiring the Security service and finally move `/config/security.yaml` to `/config/{prod,dev}/security.yaml` or use the following repo (might not work as an application, but gives the error): 

[https://github.com/BernhardWebstudio/rector-symfony-security-error](https://github.com/BernhardWebstudio/rector-symfony-security-error)

**Comment by TomasVotruba** [comment]
Thanks for looking into this.

> Apparently, rector may not be able to decide which environment it's run in.

The Rector Symfony bridge tries to locale your Kernel class. If the class is found, the Rector will try to create Kernel from it, to get to the container services, here:

https://github.com/rectorphp/rector/blob/9eb686feb76c234cbd755d97ec4fd320361eda78/packages/Symfony/src/Bridge/DependencyInjection/ContainerFactory.php#L60-L63

It uses `test` as default value for environment now.

What happens when you choose `dev` envorinment there?

**Comment by BernhardWebstudio** [comment]
Oh, right, sorry, thank you very much. Indeed, moving `/config/security.yaml` to `/config/test/security.yaml` or manually changing the mentioned [line](https://github.com/rectorphp/rector/blob/9eb686feb76c234cbd755d97ec4fd320361eda78/packages/Symfony/src/Bridge/DependencyInjection/ContainerFactory.php#L60) does indeed fix the problem too. 

I am asking myself now why the environment variable is not interpreted correctly, but I guess this has something to do with difference in environment of user executing from terminal vs. user executing the server.

On a side note, a PHP question if I may: in the mentioned [line](https://github.com/rectorphp/rector/blob/9eb686feb76c234cbd755d97ec4fd320361eda78/packages/Symfony/src/Bridge/DependencyInjection/ContainerFactory.php#L60), where is the variable `$options` coming from?

**Comment by TomasVotruba** [comment]
Could you send the fix for "dev" environment?

My assumption during writing of this code is incorrect.
"Tests" is rather extra requirement, but everyone has "dev" by default.

To your question: that whole part is wrong. It can be removed. Could you include that in PR too?

Thank you

**Comment by TomasVotruba** [comment]
Closed by #1331
